software/firmware/europi.py

```diff
@@ -213,16 +213,26 @@ def percent(self, samples=None):
         """Current voltage as a relative percentage of the component's range."""
         # Determine the percent value from the max calibration value.
         reading = self._sample_adc(samples)
-        max_value = max(reading, INPUT_CALIBRATION_VALUES[-1])
-        return reading / max_value
+        max_value = max(
+            reading - INPUT_CALIBRATION_VALUES[0],
+            INPUT_CALIBRATION_VALUES[-1] - INPUT_CALIBRATION_VALUES[0],
+        )
+        return max((reading - INPUT_CALIBRATION_VALUES[0]) / max_value, 0.0)

```

Comment:
the calculation of ```reading - INPUT_CALIBRATION_VALUES[0]``` is made twice, I'd suggest making the calculation between lines 215 and 216 with something like
```offset = reading - INPUT_CALIBRATION_VALUES[0]``` and then refer to that offset